### ISSUE TITLE
Clarify role of the Column button on tables

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#7319]
## Description
On the Pools page (and likely many other tables... but maybe not) the columns button is not really clear about what it does. You kinda have to click on it to figure out. Let's make it better.

## Screenshot

![Image](https://user-images.githubusercontent.com/101275838/254681708-7851a040-9f4b-4efd-82d8-72dbeee97f17.png)

## 🌟 Acceptance Criteria

- [ ] On table component change "Columns" button label to say: "Show or hide columns"
- [ ] In Table Columns modal, Replace "Visible columns" legend text with "Show or hide table columns"

## 🕵️‍♀️ Details

💻**AT Configuration** 
JAWS with Chrome on Windows

**Tester Feedback**

-"I cannot find the ID and Owner column toggle."

Related WCAG Criteria

- 4.1.2 Name, Role, Value: For all [user interface components](https://w3c.github.io/wcag/understanding/name-role-value#dfn-user-interface-component) (including but not limited to: form elements, links and components generated by scripts), the [name](https://w3c.github.io/wcag/understanding/name-role-value#dfn-name) and [role](https://w3c.github.io/wcag/understanding/name-role-value#dfn-role) can be [programmatically determined](https://w3c.github.io/wcag/understanding/name-role-value#dfn-programmatically-determined); states, properties, and values that can be set by the user can be [programmatically set](https://w3c.github.io/wcag/understanding/name-role-value#dfn-programmatically-set); and notification of changes to these items is available to [user agents](https://w3c.github.io/wcag/understanding/name-role-value#dfn-user-agent), including [assistive technologies](https://w3c.github.io/wcag/understanding/name-role-value#dfn-assistive-technology).

## 🧪Testing Steps

1. Access the pools table from the admin UI using NVDA
2. Navigate using the keyboard to the Columns button.
3. Determine what the role of the columns buttons is based on what you hear from NVDA.
